### ISSUE TITLE
Views

### ISSUE THREAD

**Issue by jimgwhit** [simple-mvc-framework/framework#274]
@LuckyCyborg and @daveismyname I know for a fact in Laravel and cake, you do not need CamelCase in the URL, right now I can:
~~~
site/pet/petlist
~~~
I am not forced to use
~~~
site/pet/petList

~~~
Or
~~~
site/pet/PetList
~~~
So why are you making it where this is only framework that will be different style URL?

**Comment by jimgwhit** [comment]
From Laravel docs
~~~
Defining The Child View

Great, our application layout is finished. Next, we need to define a view that contains a form to create a new task as well as a table that lists all existing tasks. Let's define this view in resources/views/tasks/index.blade.php, which will correspond to the index method in our TaskController.
~~~
Notice
~~~
resources/views/tasks/index.blade.php
~~~
small letters.

**Comment by LuckyCyborg** [comment]
It is much more simple to automatically calculate the Controller's **ViewsPath**, trying to inflect at minimum the prepared path.

And I like this style, to be honest. :smirk:

**Comment by LuckyCyborg** [comment]
@daveismyname Working with **\Smvc\Core\View**, i arrive to idea that will be very useful to rename the legacy static method **\Smvc\Core\View::render** as **renderView**

This will permit to use the method name **render** in the new/enhanced part of the Class and will be very consistent with other static methods for rendering. You have now:

renderModule
renderTemplate
renderLayout

so, why not to have even **renderView** ?

Thoughts?

**Comment by jimgwhit** [comment]
@daveismyname and @LuckyCyborg I do not use hooks or assets.php, I include headers the regular way.  So in each view folder, like the Views\dog folder I could have an inc folder for each view folder right? then say in for the  Views\dog\view.php file I could 
```
include 'inc\hdviewdog.php';
```
Is that not right?  
And I have no idea where you get that content thing from what is that?  I write out regular html / php pages with the content like: 
```
<div id="tbl-container">
    <table class="dogtable">
        <tr>
            <!--<th style="width:100px; text-align: center;">Dogpic</th>-->
            <th class="imgtd">Dogpic</th>
            <th style="width:50px;">Name</th>
            <th style="width:10px;">Sex</th>
            <th class="dogcomment">Comments</th>
        </tr>
        <?php
        if ($data['dogs']) {
            foreach ($data['dogs'] as $row) {
                echo "<tr>";
                //echo "<td>" . $row->petid . "</td>";
                if (empty($row->dogpic)) {
                    echo '<td class="imgtd">' . '&nbsp;' . '</td>';
                    //echo '<td style="width:100px;">' . '&nbsp;' . '</td>';
                } else {
                    $linev = '';
                    $linev = '<td class="imgdt"><a href="' . DIR . 'dog/view?dogid=' . $row->dogid . '">';
                    //$linev = '<td style="width:100px;"><a href="' . DIR . 'dog/view?dogid=' . $row->dogid . '">';
                    $linev = $linev . '<img src="' . DIR . 'upload/imgdogs/' . $row->dogpic . '" alt="" class="image"></a></td>';
                    echo $linev;
                }
                echo "<td style=\"width:50px;\">" . $row->dogname . "</td>";
                echo "<td style=\"width:10px;\">" . $row->sex . "</td>";
                echo "<td class=\"dogcomment\">" . $row->comments . "</td>";
            }
        }
        ?>
    </table>
</div>

<?php
echo '<table style="width:80%;">';
echo '<tr>';
echo '<td>' . $data['pageLinks'] . '</td>';
echo '</tr></table>';
echo '<br>';
?>

```
Are you trying to say I can't even write out regular pages anymore like above example.  Don't refer me to somewhere, give good example here.

**Comment by LuckyCyborg** [comment]
@jimgwhit Using:

```php
include 'inc'.DS.'hdviewdog.php';
```

DS for being portable, because \ is not DIRECTOR_SEPARATOR in Linux and BSD.

Yet, i believe that this thing is perfectly correct.

But, if you insist to call your **app/Views/dog** View files, you should use on rendering a $path starting with character **/**

for example, old syntax:
```
View::render('dog/view', $data, $error); 
```
new syntax:
```
View::render('/dog/view', $data, $error); 
```
Notice the starting with character: **'/'** as being an absolute path.

If you not do that, the new style for rendering is:
```
View::render('view', $data, $error); 
```
And will render the file:
```
/app/Views/Dog/view.php
```

IF you Controller is on:
```
/app/Controllers/Dog.php
```

**Comment by jimgwhit** [comment]
@LuckyCyborg Yes my controller will be app\Controllers\Dog.php  
The view for editing a dog will be app\Views\dog\edit.php  
So in the app\Views\dog folder I can create app\Views\dog\inc  
Then I can in edit.php for example
```
include 'inc'.DS.'hdeditdog.php';
```
For the edit header that goes with my edit view.
Now does that look correct? Or would I leave off the DS in this case?  
So to render the edit view it would be
```
View::render('edit', $data, $error); 
```
That is if I am in the Dog controller and I am rendering the edit page for the dogs, does that look right?  
Where should I call the js from, can I make a new folder in the app/Views folder named js like
```
app/Views/js
```
and top of view
```
<script type="text/javascript" src="../js/jquery.js"></script>   
```
Or how would you do it, all my view files doesn't need a js include.

**Comment by LuckyCyborg** [comment]
@jimgwhit Like I said in another post, now the MVC engine automatically calculate the Controller's Views Path. Think  that every Controller have its own Views Path, instead of the base one (/app/Views)

For the controller app\Controllers\Dog.php , the View expect to find the files in:

```
app\Views\Dog
```

Notice the name case of directories.

**Comment by jimgwhit** [comment]
@LuckyCyborg  you said  
For the controller app\Controllers\Dog.php , the Smvc\Core\View expects to find the files in:

app/Views/Dog  
I understand that but within that folder I have my view files for example
~~~

app/Views/Dog/index.PHP
app/Views/Dog/edit.php
app/Views/Dog/add.php
Etc.
~~~
Are you saying the view files have to start with a capital letter also,  
Edit.php instead of edit.php

**Comment by jimgwhit** [comment]
@LuckyCyborg earlier you used a laravel example there view folders are small letters. What is up with that?

**Comment by LuckyCyborg** [comment]
@jimgwhit While you respect the Folders name case and it match with where the Engine looks from files, the Views filename should literally respect the call, for example for your View files, you can use for:
```
app/Views/Dog/index.PHP
app/Views/Dog/edit.php
app/Views/Dog/add.php
```
the following **View::render()** calls:
```
View::render('index', $data);
View::render('edit', $data);
View::render('add', $data);
``` 
It's simple, right?

Also, if you want to see my latest pull-request and see my examples & Co.

https://github.com/LuckyCyborg/simple-mvc-framework/tree/3.0-improved-rendering

**Comment by jimgwhit** [comment]
@LuckyCyborg how does this same thing look in the new techniques you mentioned? But calling header my way, I enjoy maintaining some control.

**Comment by LuckyCyborg** [comment]
@jimgwhit The Template rendering is as usual. You can call:

```php
View::renderTemplate('header', $data);
View::render('edit', $data);
View::renderTemplate('footer', $data);
```

OR, in a new way, using the Layouts, as in my examples:

```php
// Render the Page using the Content fetching and the Layout.
$content = View::render('edit', $data, false, true);

View::renderLayout('dogs', $content, $data);
```

**Comment by jimgwhit** [comment]
@LuckyCyborg  you said 
~~~
View::renderLayout('dogs', $content, $data);
~~~
Did you mean 
~~~
View::renderLayout('Dog', $content, $data);
~~~ 
Thought it had to match controller name. I would be in Dog.PHP controller in your example.

**Comment by LuckyCyborg** [comment]
@jimgwhit I mean, if you have a custom Layout in 
```
/app/Templates/Default/Layouts/dogs.php
```
you can render it with the command:
```php
View::renderLayout('dogs', $content, $data);
```

Practically, you have Layouts example in my tree, and is very simple: it's a Page where somewhere you have:
```php
echo $content;
```
Think about a Header and Footer merged together.

**Comment by jimgwhit** [comment]
@LuckyCyborg  When this phase is complete, how about put up a simple crud example showing the latest techniques. Not that entity junk, but regular selecting data from database and paginate results using these latest techniques.

**Comment by LuckyCyborg** [comment]
After, we'll see. I should finish first "my orders". :smirk: 

You remember? I'm there because it is Job Task.

**Comment by LuckyCyborg** [comment]
@jimgwhit BTW, where is @daveismyname ?

My current pull-request already arrived at **72** commits!

**Comment by jimgwhit** [comment]
@LuckyCyborg what is an example of $content when looping through database records ln a list view with pagination? This $content has no example based on database yet.

**Comment by LuckyCyborg** [comment]
@jimgwhit The $content is just a string who represent the Page content, captured from the previous rendering command, **View::render()**. This method, now support a **fetch** mode, when it capture and return the rendering output.

To have an idea overall, in my latest pull-request, arrived at **90 commits**, I introduced a new Layout, called **legacy**.php, who is rather simple:

```
require 'partials/header.php';

echo $content;

require 'partials/footer.php';
``` 

Yep, it use a footer/header files from a sub-directory, called **partials**. The same header and footer files are used now by **View::renderTemplate**. So, if you want Layout Partials or to mix the old and new styles, the directory for them is, for example:
```
/app/Templates/Default/Layouts/partials/
```
